### PR TITLE
Add `docs` section to pull request template

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

We are able to add further documentation to the release notes by parsing
an additional block at pull requests, for example:

```docs
- Description https://mylink.com
```

This commit adds this block to the pull request template.

**Which issue(s) this PR fixes**:

Relates to https://github.com/kubernetes/sig-release/issues/667, https://github.com/kubernetes/release/pull/741

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -42,3 +42,10 @@ Enter your extended release note in the block below. If the PR requires addition
 ```release-note
 
 ```
+<!--
+**Additional documentation, like links to enhancement proposals (KEP)**:
+Please use the following format for linking documentation: `- [DESCRIPTION] [LINK]`
+```docs
+- Relevant KEP https://github.com/kubernetes/enhancements/...
```
Comment (nikhita) [inline .github/PULL_REQUEST_TEMPLATE.md]: Haven't actually seen the code diff in k/release yet, but would it be useful to have a delimiter between `[DESCRIPTION]` and `[LINK]` to come to know when the description ends and when the link begins? Kinda rare, but I'm wondering what would happen if a description has a link... :thinking: 

Also, can we have multiple links for one description? I'm asking this from a completely noob perspective :) .. for example, I wouldn't know if I'm allowed to do a `Related doc PRs <link> <link> <link>`.

Comment (saschagrunert) [reply .github/PULL_REQUEST_TEMPLATE.md]: I created a follow up PR for stripping away separators like `-` and `:` here: https://github.com/kubernetes/release/pull/748 This should solve the delimiter problem :)

Hm, multiple links to a single entry are currently not supported, for that we would have to create a new line, which equals a new entry. What works is omiting the description, like:
```docs
- https://google.com
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -42,3 +42,10 @@ Enter your extended release note in the block below. If the PR requires addition
 ```release-note
 
 ```
+<!--
+**Additional documentation, like links to enhancement proposals (KEP)**:
```
Comment (cblecker) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should this line be a part of the comment too?

Comment (saschagrunert) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think so, adding further documentation is not an optional step. :)

Comment (cblecker) [reply .github/PULL_REQUEST_TEMPLATE.md]: The problem is, because it's inside the comment, nobody will see it. The comment is only for instructions.

Comment (saschagrunert) [reply .github/PULL_REQUEST_TEMPLATE.md]: Alright, changed it that we only comment the actual instructions.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -42,3 +42,14 @@ Enter your extended release note in the block below. If the PR requires addition
 ```release-note
 
 ```
+
+**Additional documentation, like links to enhancement proposals (KEP)**:
+
+<!--
+Please use the following format for linking documentation or pass the
+section below: `- [DESCRIPTION]: [LINK]`
```
Comment (cblecker) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
section below:
- [DESCRIPTION]: [LINK]
```

**Review by cblecker — COMMENTED** [review]
What do folks do if they don't have any additional docs? Are additional docs only applicable if it's a release-noted PR?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (saschagrunert) [reply .github/PULL_REQUEST_TEMPLATE.md]: Changed as suggested.

**Comment by saschagrunert** [comment]
> What do folks do if they don't have any additional docs? Are additional docs only applicable if it's a release-noted PR?

If there are no docs they can either remove the section of leave it blank. For now the release notes tool is the only consumer of this section, so it is bound to the notes. But later on we could consume this documentation section by other tools as well, so I would keep the flexibility there. :) WDYT?

**Comment by justaugustus** [comment]
Great addition, @saschagrunert!
/lgtm
/approve

As an aside, I wonder if we can/should force docs on a PR e.g. `kind/feature` must have docs.
Well I'm sure it's possible, I'm just wondering if it's feasible...

**Comment by cblecker** [comment]
> If there are no docs they can either remove the section of leave it blank. For now the release notes tool is the only consumer of this section, so it is bound to the notes. But later on we could consume this documentation section by other tools as well, so I would keep the flexibility there. :) WDYT?

This guidance is missing from the instructions. We need to advise folks what to do in these use cases. We should explicitly state that it's only needed if you provide a release note, otherwise, leave it blank.

**Comment by saschagrunert** [comment]
> > If there are no docs they can either remove the section of leave it blank. For now the release notes tool is the only consumer of this section, so it is bound to the notes. But later on we could consume this documentation section by other tools as well, so I would keep the flexibility there. :) WDYT?
> 
> This guidance is missing from the instructions. We need to advise folks what to do in these use cases. We should explicitly state that it's only needed if you provide a release note, otherwise, leave it blank.

I added a notice about leaving it blank if there are no release notes provided.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -42,3 +42,16 @@ Enter your extended release note in the block below. If the PR requires addition
 ```release-note
 
 ```
+
+**Additional documentation, like links to enhancement proposals (KEP)**:
```
Comment (justaugustus) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -42,3 +42,16 @@ Enter your extended release note in the block below. If the PR requires addition
...
+
+<!--
+This section can be blank if you don't provide a release note.
```
Comment (justaugustus) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
This section can be blank if this pull request does not require a release note.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -42,3 +42,16 @@ Enter your extended release note in the block below. If the PR requires addition
...
+Please use the following format for linking documentation or pass the
+section below:
+- [DESCRIPTION]: [LINK]
```
Comment (justaugustus) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
@saschagrunert -- Coming back to this with a few nits that I've made as code suggestions.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (saschagrunert) [reply .github/PULL_REQUEST_TEMPLATE.md]: :+1:

**Comment by saschagrunert** [comment]
> @saschagrunert -- Coming back to this with a few nits that I've made as code suggestions.

Thanks, I changed it like you suggested. :pray:

**Comment by justaugustus** [comment]
Thanks!
/lgtm
/approve

**Comment by cblecker** [comment]
/lgtm
/approve
/hold

This looks good.

@justaugustus Would you mind sending a note to k-dev about this? Feel free to pull the hold when you're ready.

**Comment by justaugustus** [comment]
@cblecker -- Will do!

**Comment by justaugustus** [comment]
Sent to k-dev, SIG Leads, SIG Release, ContribEx, and Docs: https://groups.google.com/d/topic/kubernetes-dev/91XfKyDyCt4/discussion

**Pull request merged**